src/vbd_store.ml

```diff
@@ -29,9 +50,16 @@ let remove vbd_uuid =
   transform_vbd_list (List.filter ((<>) vbd_uuid))
 
 let get_all () =
-  Lwt_unix.file_exists Consts.vbd_list_file >>= fun exists ->
+  (* Nothing should delete the vbd_list_file, so we do not have to use a
+     Lwt.catch block here to prevent races where the file gets deleted after we
+     check that it exists but before we use it. If it does get deleted, then it
+     is fine to fail here, because that was done by an external program and is
+     an error that the user/admin should fix. *)
+  Lwt_unix.file_exists vbd_list_file >>= fun exists ->
   if exists then
     Lwt_mutex.with_lock m (fun () ->
-        Lwt_io.lines_of_file Consts.vbd_list_file |> Lwt_stream.to_list)
+        Lwt.catch (fun () -> Lwt_io.lines_of_file vbd_list_file |> Lwt_stream.to_list)
```

Comment:
`<pedantic>`Put the function on a new line as the catches above`</pedantic>`

Reply:
You're completely right, I wanted to do that but had to deal with something else so I forgot about it :smile: .